src/operations.rs

```diff
@@ -431,6 +431,31 @@ pub fn compatible_archs() -> Result<Vec<String>> {
     }
 }
 
+// which nightly channels are compatible with the current system
+fn compatible_nightly_channels() -> Result<Vec<String>> {
+    let archs: Result<Vec<String>> = compatible_archs();
```

Comment:
I think if you just terminate this line with a `?` then you can drop the whole `if` `else` construction later in the function.